misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/BallerinaStackFrame.java

```diff
@@ -128,13 +135,41 @@ private static String getStackFrameName(StackFrameProxyImpl stackFrame) {
                 return stackFrame.visibleVariableByName(STRAND_VAR_NAME) != null ? frameName :
                         FRAME_TYPE_START + FRAME_SEPARATOR + frameName;
             } else {
-                return stackFrame.location().method().name();
+                return getFilteredStackFrame(stackFrame);
             }
         } catch (Exception e) {
             return FRAME_TYPE_ANONYMOUS;
         }
     }
 
+    /**
+     * Derives filtered ballerina stack frame name from the given java stack frame instance.
+     *
+     * @param stackFrame JDI stack frame instance
+     * @return filtered ballerina stack frame name
+     */
+    private static String getFilteredStackFrame(StackFrameProxyImpl stackFrame) throws JdiProxyException {
+        String stackFrameName = stackFrame.location().method().name();
+        LocalVariableProxyImpl selfVisibleVariable = stackFrame.visibleVariableByName(SELF_VAR_NAME);
+        if (selfVisibleVariable == null) {
+            return stackFrameName;
+        }
+        Value stackFrameValue = stackFrame.getValue(selfVisibleVariable);
+        if (isService(stackFrameValue)) {
+            if (stackFrameName.equals(METHOD_INIT)) {
+                return BVariableType.SERVICE.getString();
+            } else {
+                // Here stackFrameName format will be `$resourceAccessor$resourceName$otherResourceParts`
+                String[] stackFrameNameParts = stackFrameName.split("\\$");
+                if (!stackFrameNameParts[1].isEmpty() && stackFrameNameParts[1].equals(ACCESSOR_DEFAULT)) {
+                    return ACCESSOR_DEFAULT;
+                }
+                return !stackFrameNameParts[2].isEmpty() ? stackFrameNameParts[2] : stackFrameName;
```

Comment:
Please check size of the `stackFrameNameParts` before accessing its elements, to avoid  `indexoutofboundsexception`

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/31279/commits/19e70a018836e09ced64ed7c8ce5c527328237f4